test/features/ui/widgets/table_test.dart

```diff
@@ -0,0 +1,257 @@
+        [
+          MockWidget(State('body2.1'), straightForwardContent: true),
+          MockWidget(State('body2.2'), straightForwardContent: true),
+          MockWidget(State('body2.3'), straightForwardContent: true),
+        ],
+      ],
+      bodyAlignment: TextAlignment.center,
+    );
+    final displayedChildren = table.children;
+
+    expect(displayedChildren, hasLength(5));
+    expect(
+      displayedChildren[0].render(),
+      equals(
+        '        head1       $verticalSpacer        head2       $verticalSpacer        head3       ',
+      ),
+    );
+    expect(
+      displayedChildren[1].render(),
+      equals(_verticalDivider(20 * 3 + 3 * 2)),
+    );
+    expect(
+      displayedChildren[2].render(),
+      equals(
+        '       body1.1      $verticalSpacer       body1.2      $verticalSpacer       body1.3      ',
+      ),
+    );
+    expect(
+      displayedChildren[3].render(),
+      equals(_verticalDivider(20 * 3 + 3 * 2)),
+    );
+    expect(
+      displayedChildren[4].render(),
+      equals(
+        '       body2.1      $verticalSpacer       body2.2      $verticalSpacer       body2.3      ',
+      ),
+    );
+  });
+
+  test('Creates expected children with right-side layout', () {
+    final table = Table(
+      head: [
+        MockWidget(State('head1'), straightForwardContent: true),
+        MockWidget(State('head2'), straightForwardContent: true),
+        MockWidget(State('head3'), straightForwardContent: true),
+      ],
+      body: [
+        [
+          MockWidget(State('body1.1'), straightForwardContent: true),
+          MockWidget(State('body1.2'), straightForwardContent: true),
+          MockWidget(State('body1.3'), straightForwardContent: true),
+        ],
+        [
+          MockWidget(State('body2.1'), straightForwardContent: true),
+          MockWidget(State('body2.2'), straightForwardContent: true),
+          MockWidget(State('body2.3'), straightForwardContent: true),
+        ],
+      ],
+      bodyAlignment: TextAlignment.right,
+    );
+    final displayedChildren = table.children;
+
+    expect(displayedChildren, hasLength(5));
+    expect(
+      displayedChildren[0].render(),
+      equals(
+        '        head1       $verticalSpacer        head2       $verticalSpacer        head3       ',
+      ),
+    );
+    expect(
+      displayedChildren[1].render(),
+      equals(_verticalDivider(20 * 3 + 3 * 2)),
+    );
+    expect(
+      displayedChildren[2].render(),
+      equals(
+        '             body1.1$verticalSpacer             body1.2$verticalSpacer             body1.3',
+      ),
+    );
+    expect(
+      displayedChildren[3].render(),
+      equals(_verticalDivider(20 * 3 + 3 * 2)),
+    );
+    expect(
+      displayedChildren[4].render(),
+      equals(
+        '             body2.1$verticalSpacer             body2.2$verticalSpacer             body2.3',
+      ),
+    );
+  });
+
+  test('Creates expected children with different cell width', () {
```

Comment:
What happens if `cellWidth < ContentSize`?

Reply:
It looks insanely ugly xD 

What happens is, that `Line`, which is the foundation for each row containing cells, uses the children width and does not apply any padding. The horizontal separators on the other side scale with `cellWidth` and are shorter than the content. 
A fix could be to pre-render the table content, which would violate the paradigm of `ContainerWidget`.
Because items could and will have different widths a check for the required width of the separators must include a complete render of all children. I'm not sure if that's worth it. 

TL;DR You shouldn't do that (trademark) (copyright) (what could possibly go wrong?)

PS: Thanks for the point out. Didn't think of this issue before. Maybe open a GitHub issue for that and let it rod?

Reply:
Wellll.... ^^

Can't we check the width after already rendering like the following:?
1. Render everything
2. Check that constraints are not violated (raise exception on violation)

Reply:
Finally I got some time for implementing the feature. You can find the tests in `scene_test.dart`

Reply:
For table: I added a test which ensures that each child has min and max width set. The enforcing of those constraints is a job for `Scene` and therefore not tested in `table_test`

Reply:
Well done!